### PR TITLE
feat(kusto): add kusto parser and highlights

### PR DESCRIPTION
**Pull request by Willem-J-an**
Experimental parser for kusto. Doesn't cover the full language, but a big improvement over nothing.

### CODE REVIEW

File: queries/kusto/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+(function_arguments (identifier) @parameter)
+(operation (identifier) @variable)
+(compound_expression (identifier) @variable)
+(binary_expression (identifier) @variable)
+(assignment (identifier) @variable)
+(property_identifier (identifier) @variable)
+(property_index (identifier) @variable)
+(sort_by (identifier) @variable)
+(range_operation (identifier) @variable)
```
Comment (lucario387) [inline queries/kusto/highlights.scm:27]: With these many `(identifier) @variable`, I suggest using one `(identifier) @variable` at the top then keeping those that capture not as variable below it

File: queries/kusto/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+[
+  ","
```
Comment (ObserverOfTime) [inline queries/kusto/highlights.scm:61]:
```suggestion
  ","
  ":"
```
";" should also be here but it probably can't be captured behind `_delimiter`.

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]
Add an injections file for comments.

```clojure
((comment) @injection.content
 (#set! injection.language "comment"))
 ```

File: queries/kusto/highlights.scm (hunk 2 above)
Comment (Willem-J-an) [reply queries/kusto/highlights.scm:61]: yea, ";" won't work as @punctuation.delimiter

File: queries/kusto/highlights.scm (hunk 1 above)
Comment (Willem-J-an) [reply queries/kusto/highlights.scm]: Done

File: queries/kusto/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+(property_identifier (identifier) @variable)
+(property_index (identifier) @variable)
+(sort_by (identifier) @variable)
+(range_operation (identifier) @variable)
+
+(pipe) @punctuation.delimiter
+[
+  (join_types)
+  (sort_keyword)
+] @constant
```
[the remark this answers is not in the document]
Comment (Willem-J-an) [reply queries/kusto/highlights.scm]: Done

File: queries/kusto/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  (sub_operator)
+  (to_operator)
+  (mv_apply_operator)
+  (sort_keyword)
+] @keyword.operator
+(string) @string
+(number) @number
+(bool) @boolean
+(null) @constant.builtin
+(comment) @comment
```
[the remark this answers is not in the document]
Comment (Willem-J-an) [reply queries/kusto/highlights.scm]: Done

File: queries/kusto/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+[
+  (type_cast_function)
+  (to_scalar_function)
+  (between_function)
+  (datatable_function)
+] @function.call
+
+(typed_parameter
+  (identifier) @parameter
+  )
```
[the remark this answers is not in the document]
Comment (Willem-J-an) [reply queries/kusto/highlights.scm]: Done

File: queries/kusto/highlights.scm [hunk 6]
```diff
@@ -1,4 +1,4 @@
-(source) @variable
+(identifier) @variable
```
Comment (ObserverOfTime) [inline queries/kusto/highlights.scm:1]: Why remove `(source)`?

File: queries/kusto/highlights.scm [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+(let_statement
+  (let_keyword) @keyword
+  (identifier) @variable)
```
Comment (ObserverOfTime) [inline queries/kusto/highlights.scm:4]:
```suggestion

(let_keyword) @keyword
```

**Review by ObserverOfTime — APPROVED** [review]

**Comment by Willem-J-an** [comment]
> Was that an accident?

Lol I thought I was supposed to close it, but I guess close = Abandon on github :smile:

**Comment by max397574** [comment]
yes closing is abandoning here

**Comment by clason** [comment]
> Lol I thought I was supposed to close it

Where on earth did you get that from?

File: lua/nvim-treesitter/parsers.lua [hunk 8]
```diff
@@ -968,6 +968,15 @@ list.kotlin = {
   maintainers = { "@SalBakraa" },
 }
 
+list.kusto = {
+  install_info = {
+    url = "https://github.com/Willem-J-an/tree-sitter-kusto",
+    files = { "src/parser.c" },
+  },
+  filetype = "kusto",
```
Comment (amaanq) [inline lua/nvim-treesitter/parsers.lua:976]: not needed if it's not different from the grammar name

Comment (Willem-J-an) [reply lua/nvim-treesitter/parsers.lua]: Thanks, fixed

**Review by amaanq — APPROVED** [review]

**Pull request merged**